### ISSUE TITLE
AnalyzerDependencyChecker must ignore Microsoft.CodeAnalysis.* references for missing dependency resolution

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#7049]
We seem to ignore all the Microsoft.CodeAnalysis.* references in VBCSCompiler (see [here]()), but don’t seem to do so for IDE live analysis (see [here]()).

We should create an `IgnorableAssemblyNamePrefixList` for `Microsoft.CodeAnalysis` and add it to our ignorable assembly name lists.

**Comment by mavasani** [comment]
This causes us to report the below false diagnostic in the IDE when an analyzer assembly references both Microsoft.CodeAnalysis.CSharp.dll and Microsoft.CodeAnalysis.VisualBasic.dll

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	IDE1003	Analyzer assembly 'C:\Users\mavasani\.nuget\packages\PlatformSpecific.Analyzer\2.0.1\analyzers\dotnet\PlatformSpecificAnalyzer.dll' depends on 'Microsoft.CodeAnalysis.VisualBasic, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' but it was not found. Analyzers may not run correctly unless the missing assembly is added as an analyzer reference as well.	AppCS		1	Active
```

The diagnostic itself is ignorable, as we don't run VB analyzers on C# projects and vice versa, but it is annoying to see.

**Comment by mavasani** [comment]
@ljw1004 you should not see this warning anymore, can you please verify? However, I was unable to repro the other issue you mentioned offline - please let me know if you have a repro machine available and I'll investigate.

**Comment by weltkante** [comment]
How do I suppress this warning? It's causing false-positives for us because we reference assemblies the analyzer doesn't see but which are available at runtime (and are not used when they aren't available). Since all of our projects need to refer to our analyzer package this creates a lot of noise when having solutions with many projects.

Right clicking the warning it offers me to add it to a supression file but when I click this it does nothing. I'm on VS 2015 Update 1 currently. Is there a way to add it manually to the supression file?

**Comment by mavasani** [comment]
You can suppress these warnings by applying a module/assembly level suppress message attribute:

```c#
[assembly: SuppressMessageAttribute("Category", "ID:FriendlyName")]
```

**Comment by weltkante** [comment]
@mavasani doesn't work either. It does work for other warnings but IDE1003 doesn't seem to be suppressable with this attribute, unless it requires different syntax than what's working for other warnings. I opened a new issue for the problem.
